src/Sentry.Maui/Internal/SentryMauiInitializer.cs

```diff
@@ -0,0 +1,111 @@
+using System.Reflection;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.Options;
+using Sentry.Protocol;
+
+namespace Sentry.Maui.Internal;
+
+internal class SentryMauiInitializer : IMauiInitializeService
+{
+    private Exception? _lastFirstChanceException;
+
+    public void Initialize(IServiceProvider services)
+    {
+        var options = services.GetRequiredService<IOptions<SentryMauiOptions>>().Value;
+        var disposer = services.GetRequiredService<Disposer>();
+
+        IDisposable disposable;
+#if ANDROID
+        var context = global::Android.App.Application.Context;
+        disposable = SentrySdk.Init(context, options);
+#else
+        disposable = SentrySdk.Init(options);
+#endif
+
+        // Register the return value from initializing the SDK with the disposer.
+        // This will ensure that it gets disposed when the service provider is disposed.
+        // TODO: re-evaluate this with respect to MAUI app lifecycle events
+        disposer.Register(disposable);
+
+        // Bind MAUI events
+        var binder = services.GetRequiredService<MauiEventsBinder>();
+        binder.BindMauiEvents();
+
+#if IOS || MACCATALYST
+        // Workaround for https://github.com/xamarin/xamarin-macios/issues/15252
```

Comment:
Possibly good candidate to go into the SentrySdk.Init for the mac/ios TFMs, once we get them in